src/NCronJob/Registry/JobDefinition.cs

```diff
@@ -22,22 +21,20 @@ private JobDefinition(
         string dynamicHash,
         JobExecutionAttributes jobPolicyMetadata)
     {
-        Type = typeof(DynamicJobFactory);
         IsTypedJob = false;
-        JobFullName = $"{typeof(DynamicJobFactory).Namespace}.{dynamicHash}";
-        JobName = dynamicHash;
+        JobFullName = $"Untyped job {typeof(DynamicJobFactory).Namespace}.{dynamicHash}";
 
         JobPolicyMetadata = jobPolicyMetadata;
     }
 
-    public Type Type { get; }
+    public string Name => CustomName ?? JobFullName;
```

Comment:
If during the registration a name was given, this is favored for any identification purposes.

Otherwise:
- Full type name for typed jobs
- "Untyped job {typeof(DynamicJobFactory).Namespace}.{dynamicHash}" for delegates